### ISSUE TITLE
Fails if no variation in input

### ISSUE THREAD

**Issue by botanize** [gaborcsardi/spark#9]
I create thousands of spark lines simultaneously and would like spark to handle the cases when no variation exists in the data. This currently fails due to ```cut```. 

```{r}
spark::spark(rep(0,7))
Error in cut.default(data, breaks = length(spark_ticks)) : 
  'breaks' are not unique
```
One solution would be to check for variation in the data, if none, return an underscore for each data point. I could probably work up a pull request if you're interested.

**Comment by gaborcsardi** [comment]
Hmmm. I guess we could have an option to decide what to do in this case. I remember that I thought about this, but could not decide what to do. 

So how about having an optional argument for the range? Then you could specify the range manually, and make the sparkline according to that.

**Comment by botanize** [comment]
Looks like you already have a test in place for constant inputs and that the problem only occurs when the inputs are all 0. This is really a bug in ```cut```, line 13 of ```cut.default```:

```{r}
breaks <- seq.int(rx[1L] - dx/1000, rx[2L] + dx/1000, length.out = nb)
```

The problem is that dx is defined as the min of the data, which is 0 when all data are 0, resulting in non-unique breaks. I reported the bug to R [16802](https://bugs.r-project.org/bugzilla/show_bug.cgi?id=16802)

I created a work-around, it's up to you if you want to incorporate it or not. https://github.com/botanize/spark
